src/ui/heatmap/topLands.tsx

```diff
@@ -0,0 +1,123 @@
+import { useState } from "react";
+import { TopLandsData } from "../../interfaces/common.interface";
+import { Link, Pagination } from "@mui/material";
+import Image from "next/image";
+import { TopLandForm } from "../../enums/common.enum";
+import TopSellsFilter, { TopSellingFilterBy } from "./topSellsFilter";
+
+interface TopLandsProps {
+  tableData: TopLandsData[];
+  title: string;
+  headers: string[];
+  form: TopLandForm;
+}
+
+export default function TopLandsUI({ tableData, title, headers, form }: TopLandsProps) {
+  // calculate the total number of pages to display table elements
+  const pageLength = 5;
+  const calculateTotalPages = () => {
+    return Math.ceil(tableData.length / pageLength);
+  };
+
+  const [numberOfPages, setNumberOfPages] = useState<number>(calculateTotalPages());
+  const [controlPageIndex, setControlPageIndex] = useState<number>(0);
+  const startIndex = controlPageIndex * pageLength;
+  const endIndex = startIndex + pageLength;
+  const currentPageData = tableData.slice(startIndex, endIndex);
```

Comment:
these consts could generate errors, maybe you want use useRefs or useState for that